workspaces/diff-engine/tests/lib/spawn.js

```diff
@@ -0,0 +1,102 @@
+const Tap = require('tap');
+const fs = require('fs');
+const path = require('path');
+const readline = require('readline');
+const { DiffEngineError } = require('../../lib');
+
+Tap.test('diff-engine.spawn', async (test) => {
+  await test.test('can diff a stream of interactions', async (t) => {
+    const interactions = fs.createReadStream(
+      path.join(
+        __dirname,
+        '..',
+        'fixtures',
+        'ergast-captures',
+        'ergast-simulated-traffic.jsonl'
+      )
+    );
+
+    const DiffEngine = require('../../lib');
+
+    const diff = DiffEngine.spawn({
+      specPath: path.join(
+        __dirname,
+        '..',
+        'fixtures',
+        'ergast-example-spec.json'
+      ),
+    });
+
+    interactions.pipe(diff.input);
+    let outputLines = bufferLines(diff.output);
+    let errorLines = bufferLines(diff.error);
+
+    let diffs = (await outputLines).map((line) => JSON.parse(line));
+    diffs.sort(([, tagsA, fingerprintA], [, tagsB, fingerprintB]) =>
+      compareString(
```

Comment:
any reason to not use a.localeCompare(b)?

Reply:
Not at all, just me being used to lazy `sortBy` and forgetting the most elegant way of writing the compare function 😅. Will fix.

Reply:
Fixed in https://github.com/opticdev/optic/pull/448/commits/41d6f76cbc66e26f4ea554aa3e270ffe16ec0d80